src/Extensions/MaintenanceLockExtension.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Symbiote\QueuedJobs\Extensions;
+
+use SilverStripe\Forms\CheckboxField;
+use SilverStripe\Forms\FieldList;
+use SilverStripe\ORM\DataExtension;
+use SilverStripe\SiteConfig\SiteConfig;
+use Symbiote\QueuedJobs\Services\QueuedJobService;
+
+/**
+ * Class MaintenanceLockExtension
+ * Adds a maintenance lock UI to SiteConfig
+ *
+ * @property SiteConfig|$this owner
+ * @package Symbiote\QueuedJobs\Extensions
+ */
+class MaintenanceLockExtension extends DataExtension
+{
+    /**
+     * @param FieldList $fields
+     */
+    public function updateCMSFields(FieldList $fields)
+    {
+        if (!QueuedJobService::config()->get('lock_file_enabled')) {
+            return;
+        }
+
+        $fields->addFieldsToTab('Root.QueueSettings', [
+            $lockField = CheckboxField::create(
+                'MaintenanceLockEnabled',
+                'Maintenance Lock Enabled',
+                QueuedJobService::isMaintenanceLockActive()
+            ),
+        ]);
+
+        $lockField->setDescription('Enable maintenance lock to prevent new queued jobs from being started');
+    }
+
+    /**
+     * @param bool $value
+     */
+    public function saveMaintenanceLockEnabled($value)
+    {
+        if (!QueuedJobService::config()->get('lock_file_enabled')) {
+            return;
+        }
+
+        if ($value && !QueuedJobService::isMaintenanceLockActive()) {
+            QueuedJobService::enableMaintenanceLock();
+        }
+
+        if (!$value && QueuedJobService::isMaintenanceLockActive()) {
+            QueuedJobService::disableMaintenanceLock();
```

Comment:
As with the previous comment - can you make these non-static state setters on a singleton instead?